### DISCUSSION TITLE
Add an option to suppress the User-Agent header

### DISCUSSION THREAD

**Discussion by targetdrone** [Kong/insomnia#4065] [Ideas] [answered]
I'm trying to use the Chef Infra API.  However, the mere presence of a User-Agent string triggers behavior in the API to return an HTML page advertising their Chef UI.

Today, Insomnia insists on inserting a User-Agent header with a custom Insomnia string.  I cannot override this through the Insomnia headers by adding an empty or blank User-Agent header, because any User-Agent header triggers the unwanted HTML.

I'd like to request some kind of option or capability to suppress the transmission of a User-Agent header.

**Comment by subnetmarco** [comment]
This is a great idea. @jackkav we can by default pre-load the "Headers" section with the User-Agent field that we would normally add whenever a new request is being generated. If the user wants to remove the user-agent or replace it, they can do that from the "Headers" section.

**Comment by jackkav** [comment]
You can also set it to empty string as a workaround
<img width="595" alt="image" src="https://github.com/Kong/insomnia/assets/3679927/042a89f2-ded4-47b9-91d8-88592e2dc762">

**Reply by targetdrone** [reply]
As I mentioned in the original issue, an empty string does not work around the problem. The behavior is triggered by the existence of the header, not its  contents.

**Reply by jackkav** [reply, marked as answer]
Thanks @targetdrone addressed in https://github.com/Kong/insomnia/pull/6109

**Comment by vcruzmj** [comment]
Question: after close the issue https://github.com/Kong/insomnia/pull/6109 and released to the version [2023.5.0](https://insomnia.rest/changelog#2023.5.0), there is no option (UI level) to add or enable the user-agent globally, and this is removed by default, and can only be added as a header per request.

 **This was the intended result of this?**

**Reply by DennyLoko** [reply]
I think we should have an option to enable/disable the user agent or even use a custom one, instead of having to set this individually for each request.

**Reply by joltman61** [reply]
In may case the lack of User-Agent in the query caused a 403 error from an AWS hosted server.  I only found by trying to copy my request into Postman where I found it to be working.  Through trial and error I found that including User-Agent was required.

**Reply by jackkav** [reply]
@vcruzmj Thanks for the feedback it wasn't our intention to impact older requests or meta requests like auth and gql introspection. I've fixed this in #6459 and will include it in the next dot release coming soon. Heres the approach we went with.

- new requests will have a user-agent added to headers defaulting to insomnia/[version] but editable
- if the header is disabled user-agent will be suppressed
- old requests will work exactly as they did and if suppression is desired the key can be added and then disabled.

@DennyLoko We considered global setting and request setting options but felt this approach was more intuitive.
Let us know if you have any further feedback.

**Reply by DennyLoko** [reply]
Hey @jackkav, thank you for fixing this!

Yes, you're right. The way it is working now is more intuitive. And since we can bypass its behavior, if anyone needs to change it, they could do so in the headers editor.
